cpp/benchmarks/fixture/nvbench_main.cpp

```diff
@@ -15,29 +15,42 @@
  */
 
 #include <benchmarks/fixture/nvbench_fixture.hpp>
-#define NVBENCH_ENVIRONMENT cudf::nvbench_base_fixture
 
 #include <nvbench/main.cuh>
 
+#include <string>
 #include <vector>
 
 // strip off the rmm_mode and cuio_host_mem parameters before passing the
 // remaining arguments to nvbench::option_parser
-#undef NVBENCH_MAIN_PARSE
-#define NVBENCH_MAIN_PARSE(argc, argv)                     \
-  nvbench::option_parser parser;                           \
-  std::vector<std::string> m_args;                         \
-  for (int i = 0; i < argc; ++i) {                         \
-    std::string arg = argv[i];                             \
-    if (arg == cudf::detail::rmm_mode_param) {             \
-      i += 2;                                              \
-    } else if (arg == cudf::detail::cuio_host_mem_param) { \
-      i += 2;                                              \
-    } else {                                               \
-      m_args.push_back(arg);                               \
-    }                                                      \
-  }                                                        \
-  parser.parse(m_args)
+void arg_handler(std::vector<std::string>& args)
+{
+  std::vector<std::string> _cudf_tmp_args;
+
+  for (std::size_t i = 0; i < args.size(); ++i) {
+    std::string arg = args[i];
+    std::cout << "Argument: " << arg << std::endl;
+    if (arg == cudf::detail::rmm_mode_param) {
+      i++; // skip the next argument
+    } else if (arg == cudf::detail::cuio_host_mem_param) {
+      i++; // skip the next argument
+    } else {
+      std::cout << "Adding argument: " << arg << std::endl;
+      _cudf_tmp_args.push_back(arg);
+    }
+  }
+
+  args = _cudf_tmp_args;
```

Comment:
```suggestion
  bool remove_next = false;
  auto const args_end = args.end();
  args.erase(std::remove_if(args.begin(), args_end, [&](const auto& arg){
    if(remove_next) {
      remove_next = false;
      return true;
    }
    if ((arg == cudf::detail::rmm_mode_param) || (arg == cudf::detail::cuio_host_mem_param)) {
      remove_next = true;
      return true;
    }
    return false;
  }), args_end);
```

Reply:
I'm not sure this implementation is clearer but I'm ok with it.
Stateful lambdas make me cringe a bit. I suppose I'm used to them being parallelizable.

Reply:
I'm also ok with an argument that performance isn't as important as clarity here, just a suggestion.

Reply:
I doubt it would be faster or clearer. Why not just use cxxopts. We already depend on cxxopts in libcudf.

Reply:
The point of this function (called by nvbench) is to filter out options from the given vector. 
Actual handling of these parameters is done in a different file (nvbench_fixture.hpp).

Reply:
Maybe nvbench should use cxxopts.

Reply:
Perhaps. I would consider that outside the scope of this PR though since that code (in `nvbench_fixture.hpp` -- not the file this comment chain is attached to) is not required to be updated for this change.